crate/utils/src/types.rs

```diff
@@ -67,25 +66,22 @@ impl Serialize for ExtraDatabaseParams {
         S: serde::Serializer,
     {
         serializer.serialize_bytes(
-            vec![
-                self.group_id.to_be_bytes().to_vec(),
-                self.key.as_bytes().to_vec(),
-            ]
-            .concat()
-            .as_slice(),
+            vec![self.group_id.to_be_bytes().to_vec(), self.key.to_vec()]
+                .concat()
+                .as_slice(),
         )
     }
 }
 
 impl<'de> Deserialize<'de> for ExtraDatabaseParams {
-    fn deserialize<D>(deserializer: D) -> Result<ExtraDatabaseParams, D::Error>
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
     where
         D: serde::Deserializer<'de>,
     {
         let bytes = <Vec<u8>>::deserialize(deserializer)?;
         let group_id = u128::from_be_bytes(bytes[0..16].try_into().unwrap());
-        let key = SymKey::from_bytes(bytes[16..48].try_into().unwrap());
-        Ok(ExtraDatabaseParams { group_id, key })
+        let key = SymmetricKey::try_from_bytes(bytes[16..48].try_into().unwrap()).unwrap();
```

Comment:
This implementation seems strange to me: won't these `unwrap()` cause panics if we try to deserialize the wrong data?

Reply:
The same goes for slicing over `bytes`: how can we be sure the `Deserializer` holds a serialized vector of at least 48 bytes?

Reply:
the file disappears from last "tagging" PR